core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala

```diff
@@ -111,15 +113,26 @@ class DynamicBrokerReconfigurationTest extends QuorumTestHarness with SaslSetup
 
     (0 until numServers).foreach { brokerId =>
 
-      val props = TestUtils.createBrokerConfig(brokerId, zkConnect)
+      val props = if (isKRaftTest()) {
+        val properties = TestUtils.createBrokerConfig(brokerId, "")
```

Comment:
Hmm, you're supposed to pass `null` for `zkConnect` in order to get the KRaft setup, not empty string. Admittedly, this could be documented better.

This would probably avoid having to mess around with a lot of the stuff that you're doing below, like voter ids, controller listener names, etc. etc. `TestUtils#createBrokerConfig` should do that for you.

The only thing you probably really need to do explicitly here in zk vs. kraft mode is initialize (or not) ZkEnableSecureAclsProp. The rest should be done automatically.